geopandas/io/file.py

```diff
@@ -480,6 +515,40 @@ def _read_file_pyogrio(path_or_bytes, bbox=None, mask=None, rows=None, **kwargs)
     if kwargs.pop("ignore_geometry", False):
         kwargs["read_geometry"] = False
 
+    # translate `ignore_fields`/`include_fields` keyword for back compat with fiona
+    if "ignore_fields" in kwargs and "include_fields" in kwargs:
+        raise ValueError("Cannot specify both 'ignore_fields' and 'include_fields'")
+    elif "ignore_fields" in kwargs:
+        if kwargs.get("columns", None) is not None:
+            raise ValueError(
+                "Cannot specify both 'columns' and 'ignore_fields' keywords"
+            )
+        warnings.warn(
+            "The 'include_fields' and 'ignore_fields' keywords are deprecated, and "
+            "will be removed in a future release. You can use the 'columns' keyword "
+            "instead to select which columns to read.",
+            DeprecationWarning,
+            stacklevel=3,
+        )
```

Comment:
Shall this deprecation be also in the Fiona path? If we want to deprecate it in `read_file`, I believe that the same needs to be triggered when using there keywords with Fiona, not only pyogrio.

Reply:
Yeah, I didn't yet do that. In general we pass through any additional `kwargs` to fiona, and this are fiona keywords. So after we enforce the deprecation / custom handling of those keywords, we would need to add an explicit check to ensure the `kwargs` don't contain those keys and raise an error for that on our side, if we want that those keywords stop working for fiona. 
So I was not sure if we should actually go out of our way to ensure that, or just still naively pass `kwargs` to fiona

Reply:
Ah, that makes sense. Keep it as it is then.